### PR TITLE
Improve ubuntu-cluster setup scripts

### PR DESCRIPTION
**Pull request by feihujiang**
This PR implements #6266.

Make a copy for each of the original scripts, and modify the copies. These copies will be used as working scripts afterwards.

### CODE REVIEW

**Comment by erictune** [comment]
@WIZARD-CXY @resouer can one of you review this first?

**Comment by WIZARD-CXY** [comment]
Thanks for improving the setup scripts @feihujiang first
I see a new function named `do_backup_clean`, but I don't see where to call this function in the diff.Could you please show it ?

**Comment by resouer** [comment]
@feihujiang  Your idea really make sense, but please fix the CI and check if you are missing to call do_backup_clean somewhere

@WIZARD-CXY Once the code is OK, please remember to test it through according to https://help.github.com/articles/checking-out-pull-requests-locally/

Community pair: resouer & WIZARD-CXY

**Comment by brendandburns** [comment]
Also needs rebase.

**Comment by feihujiang** [comment]
I encounted a problem while uploading the patch because of the GFW, I will try to upload it after the weekend.

**Comment by feihujiang** [comment]
@erictune , It seems the CI failure isn't caused by this script, can you recheck the PR. I have modified the script as instructions by @WIZARD-CXY  and @brendandburns .

File: cluster/ubuntu-cluster/configure.sh [hunk 1]
```diff
@@ -115,6 +150,7 @@ if [ "$etcdVersion" != "2.0.0" ]; then
 	exit 1
 fi
 
+do_backup_clean()
```
Comment (WIZARD-CXY) [inline cluster/ubuntu-cluster/configure.sh]: You can not call `do_backup_clean` like this , just do_backup_clean

**Comment by WIZARD-CXY** [comment]
I tested your new script and found that the etcd configuration is not updated because you call do_backup_clean before configEtcd function.And another shell function error @feihujiang

**Comment by feihujiang** [comment]
@WIZARD-CXY , I modified the script, Could you review it again please? Thanks!

**Comment by WIZARD-CXY** [comment]
@feihujiang The new code seems logical.Have you tested it?

**Comment by feihujiang** [comment]
@WIZARD-CXY  Yes, I have tested it, and it works!

**Comment by WIZARD-CXY** [comment]
@feihujiang OK 
LGTM

**Comment by feihujiang** [comment]
@erictune  @thockin Could you review this PR please.

**Comment by erictune** [comment]
LGTM

**Pull request merged**